src/base/bittorrent/session.cpp

```diff
@@ -141,17 +141,66 @@ namespace
         return true;
     }
 
+#ifdef Q_OS_WIN
+    // Remove this after 4.2.5 && if at least one month has passed from v4.2.3
+    // Check the commit that introduced this function and identify all other pieces of code that
+    // need removal alongside this one.
+    void fixBrokenSavePath(QByteArray &data, lt::bdecode_node &root, ResumeDataSavingManager *resumeDataSavingManager)
+    {
+        const QString path = fromLTString(root.dict_find_string_value("save_path"));
+        const int index = path.indexOf(QLatin1String("//"));
+        if (index < 1)
+            return;
+        const QString goodPath = path.mid(index).replace('/', '\\');
+        lt::entry entry(root);
+        entry["save_path"] = goodPath.toStdString();
+
+        const auto rawView = root.dict_find_string_value("info-hash");
+        const QByteArray rawHashView = QByteArray::fromRawData(rawView.data(), rawView.length());
+        const QString hexHash = QString::fromLatin1(rawHashView.toHex());
+
+        data.clear();
+        lt::bencode(std::back_inserter(data), entry);
+
+        const QString filename = QString("%1.fastresume").arg(hexHash);
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))
+    QMetaObject::invokeMethod(resumeDataSavingManager
+        , [resumeDataSavingManager, filename, data]() { resumeDataSavingManager->save(filename, data); });
+#else
+    QMetaObject::invokeMethod(resumeDataSavingManager, "save",
+                              Q_ARG(QString, filename), Q_ARG(QByteArray, data));
+#endif
+
+        lt::error_code ec;
+#if (LIBTORRENT_VERSION_NUM < 10200)
+        lt::bdecode(data.constData(), (data.constData() + data.size()), root, ec);
+#else
+        root = lt::bdecode(data, ec);
+#endif
+    }
+#endif
+
+#ifdef Q_OS_WIN
```

Comment:
I would define something related to issue (to help further understanding of the code) and then use it. E.g.:
```c++
#ifdef Q_OS_WIN
// Here is short description of the issue
#define NEED_TO_FIX_BROKEN_PATH
#endif
```

Reply:
I think I addressed this now.

Reply:
Sorry, I still see many `#ifdef Q_OS_WIN` across the code instead of some special/meaningful macro.

Reply:
Done.